segmentation/utils/spatial_analysis_utils.py

```diff
@@ -33,9 +33,40 @@ def calc_dist_matrix(label_map):
     return dist_mats_xr
 
 
-def compute_close_cell_num(patient_data_markers, label_idx, thresh_vec,
-                           dist_mat, marker_num, dist_lim):
+def get_cell_labels(analysis_type, pheno=None, patient_data=None,
+                    thresh=None, patient_data_markers=None, cell_labels=None, col=None):
+    """Based on the type of the analysis, finds positive labels that the current match phenotype or identifies cells
+    with expression values for the current maker greater than the marker threshold.
+
+    Args:
+        analysis_type: type of analysis, either Cell Label or Threshold
+        pheno: the current cell phenotype
+        patient_data: data for the current patient
+        thresh: current threshold for marker
+        patient_data_markers: expression data for column markers for current patient
+        cell_labels: the column of cell labels for current patient
+        col: the current marker
+    Returns:
+        mark1poslabels: a list of all the positive labels"""
+
+    if analysis_type == "Cell Label":
+        # Subset only cells that are of the same phenotype
+        pheno1posinds = patient_data["FlowSOM_ID"] == pheno
+        # Get the cell labels of the cells of the phenotype
+        mark1poslabels = patient_data.iloc[:, 1][pheno1posinds]
+    else:
+        # Subset only cells that are positive for the given marker
+        marker1posinds = patient_data_markers[col] > thresh
+        # Get the cell labels of the positive cells
+        mark1poslabels = cell_labels[marker1posinds]
+    return mark1poslabels
+
+
+def compute_close_cell_num(dist_mat, dist_lim, num, analysis_type,
+                           patient_data=None, patient_data_markers=None, pheno_codes=None,
```

Comment:
lets change patient_data and patient_data_markers to be just data. Then, whichever analysis we're doing, we'll still pass the data in with the data keyword, and send it to the appropriate analysis.